iscript/src/iscript/hardened_sign.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+"""Functions to sign packages with hardened runtime"""
+
+import asyncio
+import logging
+import os
+import shutil
+from glob import glob
+from pathlib import Path
+
+from iscript.autograph import sign_langpacks
+from iscript.exceptions import IScriptError
+from iscript.mac import (
+    create_pkg_files,
+    copy_pkgs_to_artifact_dir,
+    download_requirements_plist_file,
+    extract_all_apps,
+    filter_apps,
+    get_app_paths,
+    set_app_path_and_name,
+    sign_omnija_with_autograph,
+    sign_widevine_dir,
+    tar_apps,
+    unlock_keychain,
+    update_keychain_search_path,
+)
+from iscript.util import get_sign_config
+from scriptworker_client.aio import download_file, raise_future_exceptions, retry_async
+from scriptworker_client.exceptions import DownloadError
+from scriptworker_client.utils import run_command
+
+log = logging.getLogger(__name__)
+
+
+async def download_signing_resources(hardened_sign_config, folder):
+    """Caches all external resources needed for a signing task"""
+    # Get unique entitlement urls
+    entitlement_urls = set()
+    for cfg in hardened_sign_config:
+        if not cfg.get("entitlements", None):
+            continue
+        entitlement_urls.add(cfg["entitlements"])
+    # Async download and set url -> file location mapping
+    url_map = {}
+    futures = []
+    for url in entitlement_urls:
+        filename = url.split("/")[-1]
+        dest = folder / filename
+        url_map[url] = dest
+        log.info(f"Downloading resource: {filename} from {url}")
+        futures.append(
+            asyncio.ensure_future(
+                retry_async(
+                    download_file,
+                    retry_exceptions=(DownloadError, TimeoutError),
+                    args=(url, dest),
+                    attempts=5,
+                )
+            )
+        )
+    await raise_future_exceptions(futures)
+    # Return map of url to file location
+    return url_map
+
+
+def check_globs(app_path, globs):
+    for path_glob in globs:
+        if not path_glob.startswith("/"):
+            raise IScriptError('ERROR: file pattern "{path_glob}" must start with "/"')
```

Comment:
If it must start with it, should we be more graceful about this and just add it if it's missing?

Reply:
Updated it so it can handle both cases